### ISSUE TITLE
LSC Smart Outdoor (IP) Camera - Hack not working

### ISSUE THREAD

**Issue by EpicLPer** [guino/Merkury1080P#42]
Heya!

Bought a LSC Smart Outdoor (IP) Camera yesterday and tried this hack, but sadly it didn't seem to work. I added the camera to Tuya and had to update it before being able to use it (which probably was a mistake in hindsight).

![image](https://user-images.githubusercontent.com/8442196/192149609-a1c3d70a-bea6-484f-955d-09eca376161f.png)

The Firmware version on it right now is 2.10.36, and the only open ports I see are 53 and 6668.

If you need anything, I'm willing to open the camera up and do have soldering skills, I've also helped to dump the Sonoff cam a few years back and then figuring some things out :)

Thanks already for your help!

**Comment by guino** [comment]
@EpicLPer If it has 2.10x firmware you need to try this one: https://github.com/guino/BazzDoorbell/issues/13 -- did you try that one already ?

**Comment by EpicLPer** [comment]
> @EpicLPer If it has 2.10x firmware you need to try this one: [guino/BazzDoorbell#13](https://github.com/guino/BazzDoorbell/issues/13) -- did you try that one already ?

Not exactly sure what I should try there from that issue? Or do you mean instructions from the repo itself?

**Comment by guino** [comment]
@EpicLPer there's a 'Process' section on that issue, just follow it and see if it works -- that's the process that has worked for 2.9.x and 2.10.x firmware to root the device and get a ppsapp file we can patch for RTSP.

**Comment by EpicLPer** [comment]
When doing the above via the URL `http://admin:056565099@10.1.0.215/proc/cmdline` it simply says "Connection Refused".

**Comment by guino** [comment]
@EpicLPer did you do it after creating a ppsFactoryTool.txt file? 2.10 firmware is know to require that file before URLs work.

**Comment by EpicLPer** [comment]
Is it possible to just leave it empty? Afaik I have one on the SD Card right now yes, but I'm not entirely sure what to do or what to put in it sorry ^^"
Those cams seem very different from other cameras so far.

**Comment by guino** [comment]
@EpicLPer this is the information regarding ppsFactoryTool.txt : if you don’t do it correctly it wont work. It may be worth trying port 8090 as well (from 4.x firmware) - it’s always a surprise with tuya firmware.

Special note for 2.10.0 firmwae: This firmware (and newer) have port 80 closed by default -- so to use the [http://admin:05656](http://admin:5656/)... links below you have to RIGHT CLICK this link: https://github.com/guino/Merkury720/raw/main/ppsFactoryTool.txt select "Save as.." and save this file to the root of the SD card. EDIT the file (avoid copy/paste the contents of it) and modify only the ssid and password as the file requires specific format to work. When the device detects the file (in the right format) it will disconnect and re-connect the wifi (to the ssid specified) and will OPEN port 80 so the [http://admin:05656](http://admin:5656/)... links work.

Special note for 4.0.x firmwae: This firmware (and newer) have port 80 closed by default -- so to use the [http://admin:05656](http://admin:5656/)... links below you have to RIGHT CLICK this link: https://github.com/guino/Merkury720/raw/main/ppsFactoryTool.txt select "Save as.." and save this file to the root of the SD card. EDIT the file (avoid copy/paste the contents of it) and modify only the ssid and password as the file requires specific format to work. When the device detects the file (in the right format) it will disconnect and re-connect the wifi (to the ssid specified) and will OPEN port 8090 so the [http://admin:05656](http://admin:5656/)... links work but you have to add :8090 to every URL, for instance: http://admin:056565099@192.168.x.x:8090/devices/deviceinfo. Depending on the device you have you may need to use the information and files from https://github.com/guino/Merkury1080P#conclusion

**Comment by EpicLPer** [comment]
Will try this once I'm home, thanks! :)

**Comment by EpicLPer** [comment]
> @EpicLPer this is the information regarding ppsFactoryTool.txt : if you don’t do it correctly it wont work. It may be worth trying port 8090 as well (from 4.x firmware) - it’s always a surprise with tuya firmware.
> 
> Special note for 2.10.0 firmwae: This firmware (and newer) have port 80 closed by default -- so to use the [http://admin:05656](http://admin:5656/)... links below you have to RIGHT CLICK this link: https://github.com/guino/Merkury720/raw/main/ppsFactoryTool.txt select "Save as.." and save this file to the root of the SD card. EDIT the file (avoid copy/paste the contents of it) and modify only the ssid and password as the file requires specific format to work. When the device detects the file (in the right format) it will disconnect and re-connect the wifi (to the ssid specified) and will OPEN port 80 so the [http://admin:05656](http://admin:5656/)... links work.
> 
> Special note for 4.0.x firmwae: This firmware (and newer) have port 80 closed by default -- so to use the [http://admin:05656](http://admin:5656/)... links below you have to RIGHT CLICK this link: https://github.com/guino/Merkury720/raw/main/ppsFactoryTool.txt select "Save as.." and save this file to the root of the SD card. EDIT the file (avoid copy/paste the contents of it) and modify only the ssid and password as the file requires specific format to work. When the device detects the file (in the right format) it will disconnect and re-connect the wifi (to the ssid specified) and will OPEN port 8090 so the [http://admin:05656](http://admin:5656/)... links work but you have to add :8090 to every URL, for instance: http://admin:056565099@192.168.x.x:8090/devices/deviceinfo. Depending on the device you have you may need to use the information and files from https://github.com/guino/Merkury1080P#conclusion

Sadly this didn't seem to work :( The Camera still connects to my IOT WiFi instead of the normal one, despite me putting my normal one into the config to check if it even seems to read the file.
The SD Card is properly detected in Tuya tho and shows up just fine, so I doubt it's an incompatible SD Card.

**Comment by EpicLPer** [comment]
Tried a different SD Card now too, sadly same result.

**Comment by guino** [comment]
@EpicLPer sorry you had no success, every now and then we have reports of issues like this where multiple SD cards and multiple attempts don't work. Many people have reported success after repartitioning/reformatting the SD Card (trying windows/linux/etc). In any case you're welcome to post the SD card contents (from your attempt) so I can review to check if there's anything wrong.

**Comment by guino** [comment]
@EpicLPer for 2.10.x firmware the ppsFactoryTool.txt file is all you need to **enable URLs** like (adjust IP accordingly):
http://admin:056565099@192.168.x.x/proc/cmdline (port 80 -- most devices)
http://admin:056565099@192.168.x.x:8090/proc/cmdline (port 8090 -- some devices)

If you can get one of the two above working then we can try adding more files to see if it can be rooted. If neither above works then there's no way to move forward (other than opening the device). 

I still would recommend you post a ZIP of your ppsFactoryTool.txt file (from the SD card) so I can review it -- a lot of times people make mistakes because windows hides file extensions, etc.

**Comment by EpicLPer** [comment]
Will try one more time, I don't have to hold the Reset button or anything right?

**Comment by EpicLPer** [comment]
Just tried again holding no buttons or anything, and it still doesn't connect to my normal WiFi and instead stays on the IoT one.

Here's the ZIP (with changed credentials): [cam_sd.zip](https://github.com/guino/Merkury1080P/files/9728154/cam_sd.zip)

**Comment by guino** [comment]
@EpicLPer your ppsFactoryTool.txt file is fine -- are you sure the SD card if FAT32 formatted ? maybe try to use the phone app to format it then copy the ppsFactoryTool.txt file to it. If the SD card isn't formatted in a compatible format the device will simply ignore it.

**Comment by EpicLPer** [comment]
Did format via the app yeah, but sadly still doesn't seem to work.

**Comment by guino** [comment]
@EpicLPer it looks like 2.10.6 is the last 2.10.x version we had rooted, They may have made more changes in this 2.10.36 to block more things -- there's no way of knowing for sure unless someone gets a firmware dump (which requires opening device + removing the flash chip to read it).

**Comment by EpicLPer** [comment]
@guino I'm "okay" when it comes to hot air soldering, but I'd have to buy some hardware to read the flash then, but would be willing to :) Always love some challenges!

**Comment by guino** [comment]
@EpicLPer if you get the firmware I can definitely look at it. Just be careful when selecting/using a flash programmer to make sure it doesn't have 5V on 3.3V pins (many CH341a programmers out there have this issue -- and there's a mod to fix them).

**Comment by EpicLPer** [comment]
Would something like this work then?

https://www.amazon.de/DollaTek-Programmierer-ch341a-Adapter-Buchse/dp/B08HP4X326/ref=sr_1_4?keywords=CH341a&qu=eyJxc2MiOiI0LjgwIiwicXNhIjoiNC4zNyIsInFzcCI6IjQuMjQifQ%3D%3D

**Comment by guino** [comment]
@EpicLPer that is basically what I have and I did have to fix the programmer like this: https://www.youtube.com/watch?v=-ln3VIZKKaE

**Comment by EpicLPer** [comment]
I also have a ST-LINK and JR Programmer 2 still laying around from other projects but doubt I can reuse them here? I'm quite a beginner with these kinda of things

**Comment by EpicLPer** [comment]
Took the camera apart now, already ordered the flasher :)

Noticed it has TX and RX on it tho, is there no possibility to connect vis serial and dump it? 
![405148E3-F82D-4625-AF2E-32285F066B9D](https://user-images.githubusercontent.com/8442196/194596218-3c5fb1ee-4732-4c73-bff4-b6c7f634b358.jpeg)
![542A9A11-872B-41A6-B42C-1F01387A98D3](https://user-images.githubusercontent.com/8442196/194596250-820e8b65-cd95-4e79-8704-569a7e5aab39.jpeg)

**Comment by EpicLPer** [comment]
I've read up on this a bit and apparently it's also able to dump such chips via an Arduino? If so could try that, or wait on proper hardware and be less likely to damage things ^^"

**Comment by guino** [comment]
@EpicLPer great pictures. The XMC chip beside the RX/TX pads is the flash chip you need to read. The RX/TX pads (along with GND at the bottom would be for TTL-UART access which can sometimes be helpful, but you would need a TTL-UART 3.3V adapter (USB or serial), or a board with serial TTL connections (i.e. Raspberry Pi, etc).

**Comment by EpicLPer** [comment]
Do have that :)

Do you have Discord or Telegram by any chance to maybe easier talk there for a but until I have some results I can post here again? Normal conversation always bloats issues up a bit.

**Comment by guino** [comment]
@EpicLPer if you have a Pi and you're willing to solder some wires to the board we can try to dump the flash to a SD card using the bootloader. Send me an email directly (my email is on my github profile) and we can exchange contacts.

**Comment by mrwiwi** [comment]
Exact same results with LSC Smart Doorbell ! Will follow your progress with attention :)

**Comment by EpicLPer** [comment]
> Exact same results with LSC Smart Doorbell ! Will follow your progress with attention :)

We already got RTSP with Audio working, but it's via a test-mode of that camera.  
You can test and see if yours is the same as mine (which we used to test this on) by creating a file called `_ak39_factory.ini` with nothing inside it on the root of the SD, and see if it screams at you in Chinese then which means it entered "Factory Test Mode".  

Nothing will work tho and it won't connect to WiFi, this is literally only to see if your camera is similarly built than mine is, so then chances are high once we release a working version yours will work with it too! Please do report your results :)

**Comment by mrwiwi** [comment]
> 

Will try when i'm home with some time, thank you !!

**Comment by qba89** [comment]
I confirm, after placing the _ak39_factory.ini file when launched, the camera screams in Chinese and does not connect to any Wi-Fi. This applies to a camera like in #1. I am waiting for a working version, and thank you for your work

**Comment by EpicLPer** [comment]
> I confirm, after placing the _ak39_factory.ini file when launched, the camera screams in Chinese and does not connect to any Wi-Fi. This applies to a camera like in 1 comment. I am waiting for a working version, and thank you for your work

That's good news!  
There's another file called `_ht_ap_mode.conf` which then tries to load a hostapd binary from the SD Card, which @guino all figured out and built a custom one that in turn then loads a script off of the card too to further load things :)  

I'll try to create a "proof of concept" script that loads RTSP up and all at some point when I have time, it's a bit hard tho cause hostapd gets killed when the camera starts the WiFi driver for whatever reason and in turn also kills all Telnet servers via Busybox we tried, so the current workaround is to modify the WiFi driver start file to simply comment out the hostapd killall line. I want to find another solution for this tho to not modify anyone's cam files internally and risking a brick.

**Comment by guino** [comment]
@EpicLPer I was in Europe this past week and was able to buy one of these cameras, so I should be able to do some more testing/tweaking to iron out the rooting/setup of this device (after I get myself some EU plug adapters).

**Comment by EpicLPer** [comment]
> @EpicLPer I was in Europe this past week and was able to buy one of these cameras, so I should be able to do some more testing/tweaking to iron out the rooting/setup of this device (after I get myself some EU plug adapters).

It's just 5V, if you have a soldering iron and a spare USB cable you could just snip it and solder it directly to the cam.

**Comment by jonesMeUp** [comment]
My colleague bought this device today. He also has 2.10.36, but the app wants to update to 4.8.8
I told him not to do the update and wait for your progress here, becaue I think a newer version will make more trouble.
So another one is awaiting a solution for breaking the china chain.

I really learned a lot from this project, but there is so much left to learn...

**Comment by guino** [comment]
@jonesMeUp we have an exploit for this device but it needs tweaking and some work before making it available. I have the device in hands so I should be able to do some testing and publish it when it's working well.

**Comment by jonesMeUp** [comment]
**Super**
Will it be for 2.10.36 only, or will there also be a hack for 4.8.8?
(I think 2.10.36 will be replaced in the store soon)
thx!

**Comment by guino** [comment]
@jonesMeUp I have 2.10.36, so that's what I'll work on first. If my device can be upgraded to 4.8 later I can look into it as well.

**Comment by mikonline** [comment]
Bought three of these cameras without doing any research before. Can't wait for this exploit :)

**Comment by guino** [comment]
So I have an interesting development on this device... I spent hours yesterday trying to replicate the hack we did on EpicLPer's device without success. Only then I realized my device is running older firmware (2.10.22) -- so I backtracked on other exploits that didn't work on his device and one of them worked on both my Outdoor camera (2.10.22 firmware) and on my Rotating camera (2.10.28 firmware).

After I had root access I noticed they changed some things between 2.10.22 and 2.10.36 that explained why the hostapd hack didn't work, and after knowing the differences I was able to make it work on my devices as well.

I would like someone with 2.10.36 (or newer) to try the below and report back if possible -- this is only for ROOTING (no RTSP/mods yet):

1. Unzip this into the root of a FAT32 SD card:
[210root.zip](https://github.com/guino/Merkury1080P/files/10026337/210root.zip)

(It should create two files: _ht_av_tuning.conf and shadow)

2. Eject the SD card properly (using eject function in windows or umount command in linux)
3. Insert the SD card on the device while POWERED OFF, then power on the device with the SD card in it
4. Let it boot normally (it should still connect to the phone app when done).

Try to telnet and/or FTP into the device with user: **root** password: **telnet** (please note this password is defined on the SD card file I provided -- it is not part of the firmware).

I will use the above information to leverage our options if it works on 2.10.36 and/or newer.

**Comment by mikonline** [comment]
It works. I am able to telnet into the camera:

☁  ~  telnet 192.168.201.77  
Trying 192.168.201.77...
Connected to 192.168.201.77.
Escape character is '^]'.

anyka login: root
Password: 
welcome to file system
[root@anyka ~]$ 
[root@anyka ~]$ 
[root@anyka ~]$ ls
bin      data     dev      etc      lib      linuxrc  mnt      proc     sbin     sys      tmp      usr      var
[root@anyka ~]$

**Comment by qba89** [comment]
Hello, first, I would like to thank you for the work you put into this, which brings results.

Core Module: V2.10.36
MCU module: V2.10.36
Telnet access
FTP access
Works from the phone app too

**Comment by guino** [comment]
@mikonline just to confirm: your firmware is 2.10.36 ?
@qba89 Thanks for posting your positive results.

**Comment by mikonline** [comment]
@guino Yes 2.10.36

**Comment by McPrapor** [comment]
Can confirm telnet works for LSC Smart Connect rotatable camera as well. Any clue how to get rtsp from it?

**Comment by guino** [comment]
@McPrapor I have a LSC rotating camera on 2.10.28 -- can you confirm the firmware on yours ?

I just finalized what we need to run our scripts from the device without needing to modify the flash or use telnet.
The next step is putting in the right set of tools in it (busybox, telnet, upload, download) and allowing to use a main application from the SD card, only then I can start looking into patch the main application to enable RTSP.

I will likely need to patch each individual version of the main application to make it work, so I'll do mine first (2.10.22 and 2.10.28) then will need a copy of other versions to make a patch for people to try. I believe I also have a copy of 2.10.36 but I'd like to confirm at least by checking md5sum.

This will definitely take a few more days of work at least.

**Comment by McPrapor** [comment]
I have 2.10.28. It offered me upgrade to some newer version, but I
rejected. Interesting fact, with the 2file fix above it says now there are
no updates available.

**Comment by guino** [comment]
So an update on this device, my outdoor camera has 2.10.22 (no update offered in tuya app). I have managed to get it booting correctly with a (heavily) patched application that has RTSP working. The same files also work on my Rotating camera running 2.10.28.

Worth mentioning:
* 2.10.22 and 2.10.28 firmware don't have the RTSP audio support so basically I'm running 2.10.36 application on the older firmware (without any issues).
* I'm trying to work a way to make both the tuya app and RTSP to work -- right now I can only use the phone app in SD/low resolution and if I try to switch to HD it fails because it is 'in use' by the RTSP server.
* I still have to do a few tweaks (disable some chinese start sound + make sure it can boot offline) and clean up the scripts before I make it available.
* There's no way to make snap/mjpeg work on these devices because they don't have a dedicated JPEG buffer like older devices (they only allocate a buffer when needed to transmit a picture to the server), so the only way to make snap/mjpeg for these devices will be using ffmpeg like here: https://github.com/guino/rtsp2jpeg

I will probably publish what I have in the next few days (even if I don't get the tuya app + RTSP working together), just wanted to provide an update.

**Comment by jonesMeUp** [comment]
wow, you did it again!
sorry to hear about the missing jpeg buffer.
i can't wait to see if i can run my common offline initrun.sh on it.

**Comment by guino** [comment]
@jonesMeUp well, the OS and file structure is totally different on these devices -- so no more boot parameters, scripts, etc. It'll be just placing some files in the SD card and booting up. I don't want to 'host' device files in github so I'm trying to find the right tools to extract the files from their update. From what I can tell they're only updating devices to 2.10.36 if your account is in the European servers, but being on 2.10.36 would only require a patch file like we've done in the past. I am still debating if there's any 'benefit' in not updating to 2.10.36 (other than not having to enroll the device on Europe's server).

**Comment by guino** [comment]
I was able to get the phone app and RTSP working together after a whole lot of trying and re-writing the video call back function by hand. What's left to do now is:
* make a cgi script to enable/disable telnet on demand (because there will not be a telnet password on this device)
* adjust logparser for motion detection/notification
* documentation/write up

**Comment by guino** [comment]
For anyone waiting on this: https://github.com/guino/LSCOutdoor1080P

**Comment by EpicLPer** [comment]
Awesome! Does audio work already for it?

**Comment by guino** [comment]
@EpicLPer yes! audio works already! Thanks for the coffee (and the help)!

**Comment by guino** [comment]
@McPrapor It is odd that you don't get the 2.10.36 update notice on the LSC app with Europe servers -- I tested just now and I still get the prompt to update the outdoor camera, so I'd expect the same with the rotating camera (basically same firmware).

To be clear you WILL NOT get the update prompt while using the files to enable RTSP (because that's already 2.10.36), so if you wish to update your device you should remove the SD card, make sure you're using the LSC app with an European account and hopefully that should be enough.

I can confirm that running 2.10.36 (patched application) on my 2.10.22 firmware seems to cause the up/down movement of the camera to move more than it does with the 2.10.22, I believe this is what's causing the camera to point almost all the way down at start up. At a quick look I was not able to find any quick setting to change that behavior.

**Comment by guino** [comment]
If any one here has a rotating camera on 2.10.36 I'd like to get the output from this telnet command:
```
cat /etc/config/_ht_sw_settings.ini
```
That may give us a hint as to fix the issue with older firmware.

**Comment by jonesMeUp** [comment]
**first things first: thanks for your time and energy!**

hm, like always all is a little bit more complicated here :(
as i said before:
> My colleague bought this device today. He also has 2.10.36, but the app wants to update to 4.8.8
>I told him not to do the update and wait for your progress here, becaue I think a newer version will make more trouble.

After guino was on it, i asked santa for a camera... so my mother bought one for me to get it to work before christmas.
I know, i talk too much when i am frustrated and filled up with beer... sorry
now some facts:
I put my camera online today. In the smartlife app, i don't get any info because when i deny the "update SDK version to V4.8.8"
it takes me out of the configuration.
Then i tried to get the camera of my colleague into my network, but after the pairing sound (i can see the device in my router), the app counts from 2:00 to 0:00 ignoring that the pairing was done.
I don't know what version i have, but my colleague told me his device was 2.10.36 and he got the same update info like me.
He said he had denied all tuya update wishes, so i think i have the same version (both of us have the device without movement).
so, our 2.10.36 seems to be different from your versions?! 
our anyka_ipc size: 4187428
your anyka_ipc size: 4258888
After i had nothing to lose, i used guinos guide to get (your) 2.10.16 patched and put it over the colleague version accepting the fact that it may brick the device. Same result: pairing sound, visible in the router - but smartlife counts to 0.
So the Hardware seems to be the same, no brick at all. but still a non working device.
I am willing to do all your crazy wishes on the device of my colleague - its already unusable, at least with tuya app, and it has still warranty left for some time...
Would be nice if you have an idea of getting the device to work without the fu*§"§% smartlife app, but no problem if the device will be declared as doorstop.

**Comment by guino** [comment]
@jonesMeUp glad you got it working. I have seen issues with chinese cameras having poor quality wifi chips, I have a camera where the wifi causes the device to freeze frequently and disabling wifi and using Ethernet (which thankfully is present) makes the device very stable. So I am not surprised you had issues with it.

@McPrapor I will add my rotating camera to my LSC app to see if I am offered the update, but it is surprising that possibly slightly different version of the device had no update available - maybe they removed it specifically because of that initial position issue we’re seeing on 2.10.36

**Comment by guino** [comment]
@McPrapor I am not being offered to update my rotating camera either. It obviously works with 2.10.36 with that start position being the only ‘noticeable’ glitch. I can tell you that 2.10.36 has a number of new settings not present on 2.10.28 so I would hope they could allow it to work correctly, but without sample settings it would be a guessing game which can take a long time to figure out.

**Comment by guino** [comment]
@McPrapor and anyone else with the rotating camera: I spent quite a bit of time (as expected) trying different values for settings (and even did some patching attempts) and was finally able to get the rotating camera to start in the correct position. I have updated the repository so it should be ready to go -- if you already have it all setup you only need to update hack.sh in the SD card. So really there should be no reason to try and get 2.10.36 on these cameras.

**Comment by McPrapor** [comment]
Oh what a wonderful news! Updated and checked my cams, looks great!

**Comment by jonesMeUp** [comment]
@guino Captain we have a problem: motion event.
Tried to make my "single file offline" solution for this device.
When the device is online, all worked well. when i cut inet - all went working.
But when i cut inet and **then** boot the outdoor - no motion event is detected anymore.
I hope thats, not again, only hitting me (after talking, again, bad about tuya devices).
I tried to find the problem in ghidra, but i was lost in hex.
Motion event must be handled in anyka_ipc_rtsp, else it wouldn't work after switching off inet.

And another think i don't understand:
> The reason why we're using a passwordless telnet is to prevent writing to the flash memory on every boot.

Where can i see when something is written to the flash? I thought /tmp is always in RAM.

I think i am too old for this :(

**Comment by guino** [comment]
I have just updated the repo's readme with how you can update the outdoor camera to 2.10.36 if it isn't being offered in the app.

@jonesMeUp I'll have to try it out and check the code to see why it isn't working -- likely the tuya code is 'waiting' for connection with the server before it fully starts its own functions. In these devices the RTSP code is separate from the tuya code so it should start up right away without internet (but the motion detection is a tuya function).

Regarding the flash issue: The start scripts mount a read-write partition under /etc/config and copies some things in there every boot if they're present in the SD card, then deletes them and it happens every boot, so if we wanted to do a telnet with password we'd have to accept that we'd be writing to the flash every boot for no good reason. While working on the rotating camera I also found that even without telnet a file gets copied from read-only flash to /etc/config on every boot so I made an 'override' to prevent that in my latest hack.sh. We can take care of all those things by writing a custom partition to the flash but I generally like to leave the device's flash untouched whener possible.

**Comment by guino** [comment]
@McPrapor with my rotating camera, no matter what position I left the camera before reboot it always boots up to the normal front-facing position (same angle). So I'm not sure why you're getting a different behavior.

**Comment by jonesMeUp** [comment]
@guino If i understand you correctly the problem is the etc folder (not the tmp?). but we used it also on the doorbell for passwd/shadow, TZ, dropbear. Is this because it was handled another way on the doorbell? And if so, can't we just place the config files also in the ram? You used the mount bind (i never ever heard off it until you placed it in your git) - can't it be used for "all configs to ram"?)
Might sounds all a bit silly, but since you crashed into my life... I must learn so much in so less time...

**Comment by jonesMeUp** [comment]
@guino thank you really much, that you try to hammer your wisdom into my head.
I am really trying to understand your info and i read more linux guides for this project than in my whole life before... 
What i **think** i have learned from your postings:
- doorbell doesn't make such symbolic links to read-write partiton, so we are not in panic about /etc
- we could make a warkaround for the /etc folder that would be destroyed by an update (but we don't want updates?!)
- the problem of the motion detection can be fixed by ghidra (but not by me)

what i don't understand:
- how can **_I_** see if its written to ram or to flash.
- where to find such stuff like motion detection in ghidra.
- why the bard's tale was more important to me than linux.

Perhaps it looks like i take a short look and then give up, but thats not true! before i post here, i have invested **at least a day** of trying. I am a poor pensioner, so i have the time.

**Comment by guino** [comment]
@jonesMeUp you are right about the symbolic link, I always worry about updates being done without express authorization and it may be possible to remove it the motion detection limitation in ghidra — haven’t looked for that specifically yet.

You can execute the `mount` command in telnet to see what is mounted — anything that is not read only and not in the SD card (/dev/mmcblk*) would be flash storage so it is ideal to be mindful of anything copied/modified in these locations.

The tuya code usually has a connection wait which is usually a while loop checking a variable or calling a function to check if the device is online (with a wait/sleep inside) before running other tuya functions (like motion detection) — most likely 3-4 lines of code that would have to be modified to bypass the wait. 

I have played my fair share of video games, in fact the only reason I learned coding and OS stuff was so I could play games on the computer. If you have time and want to learn I praise your efforts and will try to answer when I have time but ghidra is basically reverse engineering which is a step further out from coding, so don’t feel bad if it takes time - I spent half a day myself yesterday to figure out 1 setting, so it takes a lot of time even experienced people to do this stuff.

**Comment by jonesMeUp** [comment]
@guino thx for your support!
Meanwhile i have found **some** places with MQTT,  NTP and Network_connection loops.
e.g:   `0009c4b0 f8 ff ff 1a  bne LAB_0009c498`   replaced by  `0009c4b0 00 f0 20 e3 nop`
But there seems to be many more. I will try the next time to get it done.

**If** i am able to fix the buggy tuya online code, i will go on with my 1 file solution and i will have many questions about the flash writings. (If i see it correctly, tuya overwrites /etc/TZ also at every boot!?)

**Comment by guino** [comment]
@jonesMeUp I had a quick look and it seems very different than previous code I've seen (though the code has the same effect) -- I haven't tested anything but you can try changing this
```
00088c94 02 00 00 0a         beq    LAB_00088ca4
```
to
```
00088c94 02 00 00 ea         b      LAB_00088ca4
```
I would expect it to initialize the tuya functions even if it can't reach the tuya servers.

**Comment by jonesMeUp** [comment]
well, thats 1 part of my changes, but they have many loops. right now i have done this, but its not enough:
Main
        00088c78 00 30 a0 e1     cpy r3,r0
        00088c78 01 30 a0 e3     mov r3,#0x1

IPC_APP_Get_Mqtt_Status
        00088c8c 00 30 a0 e1     cpy r3,r0   
        00088c8c 00 30 a0 e3     mov r3,#0x0

NTP
        00088cac 00 30 a0 e1     cpy r3,r0
        00088cac 00 30 a0 e3     mov r3,#0x0

ht_network_get_connect_state
        00088dbc 00 30 a0 e1     cpy r3,r0
        00088dbc 01 30 a0 e3     mov r3,#0x1

Main
        00088e6c 85 ff ff 1a     bne LAB_00088c88
        00088e6c 00 f0 20 e3     nop

Main
        0009c4b0 f8 ff ff 1a     bne LAB_0009c498
        0009c4b0 00 f0 20 e3     nop

**Comment by icecream4all** [comment]
forgive my ignorance but I'm wonderin how to download video files remotely.

**Comment by jonesMeUp** [comment]
welcome icecream4all,
the video files are saved in a non standard format to the sd-card.
If you are running a raspi as smarthome central (like me) is the best to let ffmpeg do the job.
When you tell us more about your environment we can give some more info.

**Comment by guino** [comment]
@jonesMeUp I will take a look at the code, not sure when I can actually test it offline. There's a built in function to play mp3 files, but unless we have a way to call the function 'on-demand' (from linux command line) there's no way to make it play what we want (I can't directly 'call' a function inside their code from linux command line).

@icecream4all Using the files I provided you can 'download' the video files by browsing to http://user:password@IP:8080/ (user and password configured in httpd.conf) then browsing into the DCIM folder and clicking on one of the .media files -- you should be able to download the file directly from a script with curl/wget or similar too. The files are (as mentioned above) in non-standard format but you can play them without audio with mplayer or VLC (if you select H264 demuxer).

**Comment by jonesMeUp** [comment]
@guino still so much left to learn. i thought your play.cgi jumps directly into the tuya binary to play the mp3.
sometimes i feel like a stoneage warrior in a raspi world. when i was in school, the zx81 was not invented. but we had already the pyramids!
> I will take a look at the code, not sure when I can actually test it offline

No problem, i have my offline version and can test my onefile. I don't care how much time it will need. Its only a try to give something back to your wonderfull project.
btw, would be nice if you could drop me a line to understand why play.cgi works on doorbell, but not on outdoor (like you would tell it a 5 year old child)

**Comment by guino** [comment]
@jonesMeUp on older firmware they had a loop checking to see if anything needed to be played — it was just checking variables in the memory. So play.cgi simply set the variables in memory and the loop would detect and play it. 

At a quick check I did not find anything similar for these newer devices, so if we can’t find it we would have to find some loop that is running but isn’t being used and basically change it out to monitor some memory location and call the mp3 playback function when we need it. The closest thing I found was the thread to play the siren alarm — we could trade that for play.cgi (which would make the siren function no longer work correctly).

It is not impossible, just a pain and time consuming to modify each individual instruction to write a whole new function without braking whatever is there.

**Comment by jonesMeUp** [comment]
@guino ok, that sounds to me like a real good idea. Have you ever tried the siren? its a bad joke! is not a siren, its just a mp3 playing. I have a fire alarm here... thats something i would call siren, but that %$%&$ outdoor whisper is called a "siren"???
so it would nice to have it replaced by a usefull function.
sorry that you have to invest so much time in me, but i hope to give it back twice into the project.

**Comment by guino** [comment]
@jonesMeUp Your changes seem ok, I'll have to do some testing when I have a chance and include them in the main patch. Meanwhile we can make an 'offline patch' with these 2 changes so anyone else can try if they want.

**Comment by McPrapor** [comment]
@guino  I dumped today fw from one similar cameras sold in local stores, where no card tricks worked. I wanted to enable telnet for beginning, as uart pins being disabled most probably in uboot. So far I was able to disassemble it only with binwalk, but apart of rootfs, there are zlib files and not all of them could be unpacked with gzip and some jffs2 files. Could you please share some tricks how to disassemble those?

**Comment by icecream4all** [comment]
@jonesMeUp I run homeassistant. My wifi is crappy at the moment which sucks cause I'm building a timelapse. The Tuya app keeps telling me there are no vids recorded (end of video) but now I can manually access the clips thanks @guino . I take some screenshots from the clips to fill the holes in my timelapse.

**Comment by guino** [comment]
@McPrapor jffs2 in a firmware dump sometimes is tricky. Basically you have to extract the jffs2 partition as a separate file, then on a linux machine you can mount it with something like on this script: https://elinux.org/Mount_jffs2 -- you may need to adjust the mtdblock0 to something else depending on what already may be present on your OS.

Basically (all on the linux machine):
-use `dd` to 'extract' the entire JFFS2 from the firmware dump into a separate file (because binwalk will try to extract separate jffs2 pieces and anything in between (i.e. zlib files)
-check to see if you have any mtd modules and devices loaded: `lsmod | grep mtd` and `ls -la /dev/mtd`
-if you already have `/dev/mtdblock0` you may need to see what's using it and/or adjust the script to use a different number not in use like mtdblock1.
-execute the script (perhaps with different mtdblock value), passing the extracted JFFS2 from the firmware -- it may throw some errors for some drivers that you don't have like mtdram, mtdchar) -- that should be ok, hopefully you'll be able to mount it to view/edit the files.

I will tell you advance jffs2 is usually for settings, so you may go thru all this and not find anything useful to gain access to the device, but you could be lucky and find a setting to turn on telnet, etc -- but you'd have to flash your changes back to use it.

**Comment by EpicLPer** [comment]
@jonesMeUp I'd love for those cams to somehow become "dumb RTSP/ONVIF cams", and this is the closest approach yet!  
There's also the OpenIPC project, but sadly Anyka is not supported properly yet.

**Comment by jonesMeUp** [comment]
@EpicLPer yep, i have no interrest in an AI spy. Just a dump camera would be nice.

**Comment by EpicLPer** [comment]
> @EpicLPer yep, i have no interrest in an AI spy. Just a dump camera would be nice.

If someone is skilled enough to try and port OpenIPC to these specific Anyka chips we may have a chance :)

**Comment by jonesMeUp** [comment]
@guino I could need an advice. When i set the "wait for ntpd sync" in the background loop:
```
####################################################################################
# Event loop 
####################################################################################
(
  while [ $(date +%s) -lt 1645543474 ]; do :; done; # Wait for synced localtime
```
The streaming folder didn't get the actual time, because the app starts before the time is synced.

When i set it in front of the loop, the camera seems to be in an endless loop (perhaps a wathdog restarts it)
I set a sleep() **within the app** in _tuya_main()_,  it doesn't change the date of the streaming folder.
```
          sleep(0x20);
          IPC_APP_Sync_Utc_Time();
```
On the device `date` shows the correct time, but stream-time is always 1h wrong and stream-folder date is  always 1970.
The timestamp on the stream will be corrected after ~10s when using the timesync-wait in the background 
and comes up with the "correct" time (1h wrong) as soon the stream is shown in vlc

**Comment by jonesMeUp** [comment]
@EpicLPer i don't think someone will do it. 
But as long as we have guino... i am **very happy** to get the 24/7 service here.
Of course we must have an eye on guino and prevent holiday/marriage/kids/joing a cult/drugs/netflix/gambling/...

**Comment by McPrapor** [comment]
Hey, folks. Any chance to get a patch to make it start offline?

**Comment by guino** [comment]
@jonesMeUp The sleep(0x20) you added is really just 32 milliseconds -- if you want to sleep for 20 seconds you should do sleep(20000) -- you may need to put that 20000 into a memory location and load the parameter from it because you probably can't load 20000 into a register directly.

Another approach could be to leave the code unchanged (which tuya uses to wait for it to be 'online' with their servers) and set the variable/memory in the process directly (so it thinks it's online), that is: once your NTP service syncs. On 2.10.36 the variable that tells if mqtt is online seems to be at 0x434d1c -- so something like this could work (untested):
`echo -en '\x01\x00\x00\x00' | dd of=/proc/550/mem bs=1 count=4 seek=$((0x434d1c))`

So basically you could try (without your offline patch) to just run the command above when your NTP service syncs the time to see if the device starts 'acting' like it's online (motion notifications, recording, etc).

**Comment by guino** [comment]
@EpicLPer I just updated the repo with a feature to control the PTZ of the camera -- I think I recall you wanted that too?

**Comment by jonesMeUp** [comment]
@guino Well, thats a very cool idea! I will try this in the evening.
@McPrapor Above your post is an offline solution. But if i were you, i would wait until the localtime is fixed.

**Comment by jonesMeUp** [comment]
@guino After many hours of ghidra surfing, i found that it looks like the app reads the hwclock, so i copied the time from system to hardware clock `hwclock -w`. In telnet it looks good, the 1h differ is gone. So its the right direction, but the app refuses to update the change. Tomorrow i will dig into it again.

**Comment by guino** [comment]
@jonesMeUp Looking at the code you can try this one (together with my first command):
`echo -en '\x01\x00\x00\x00' | dd of=/proc/$PID/mem bs=1 count=4 seek=$((0x47b7a0))` -- this seems to be what gets that other loop to stop, and it does some stuff with Time zone and system clock, so patching it may not be a good idea (but this should allow the normal code to execute).

**Comment by jonesMeUp** [comment]
@guino well, that fix breaks the motion event. 
The only problem i have ist that the streamingfolder thinks its 1970. I wasted again much time on this topic.
I switched the internal logging of the cam on to get a little more info. I will look tomorrow if i find something there to explain it. 

I think its a timing problem, because the stream neeeds also a few seconds to sync the date. 
The app is starting to fast for ntpd to end this job and the streaming folder doesn't sync the date after init.

**Comment by guino** [comment]
@jonesMeUp for these devices, have you tried to sync the time in hack.sh (and wait) instead of running the event loop in the background ? you will have to adjust the code that changes the memory directly (so that waits for the main application to be running), but I would expect the main application to start with the correct date. The only question is 'how long' we can wait to run the daemon and anyka_ipc before any type of watchdog restarts the device.

**Comment by jonesMeUp** [comment]
@guino yep, long before i put the waiting into the bg which seems to trigger a watchdog. there is no chance to wait for so long.
For a test i put **iirc** 8 * `nop` instruction at the start of  `void ht_log_init(void)` the first sub in main().
It worked without killing the log function, but we could live without the logging... 
My problem is the `ldr` part. i don't understand how to load the statusMQTT variable into `r3` in assembler code.
i "copied" the code block from tuya_main() 
```
          while (statusMQTT != 1) {
            sleep(1);
          }
```
```
        0009c498 01 00 a0 e3     mov        r0,#0x1
        0009c49c 18 de fe eb     bl         <EXTERNAL>::sleep
        0009c4a0 a0 30 9f e5     ldr        r3,[PTR_DAT_0009c548]
        0009c4a4 03 30 8f e0     add        r3,pc,r3
        0009c4a8 00 30 93 e5     ldr        r3,[r3,#0x0]=>statusMQTT
        0009c4ac 01 00 53 e3     cmp        r3,#0x1
        0009c4b0 f8 ff ff 1a     bne        LAB_0009c498
```
into the free part of `void ht_log_init(void)` but i didn't get ldr to point to the statusMQTT.

**Comment by guino** [comment]
@jonesMeUp here's what I'd try (this will SET the statusMQTT and at the same time skip the loop):
```
  0009c4a0 a0 00 9f e5         ldr    r0,[0x9c548]
  0009c4a4 01 30 a0 e3         mov    r3,#0x1
  0009c4a8 00 30 00 e5         str    r3,[r0,#0x0]
  0009c4ac 01 00 53 e3         cmp    r3,#0x1
  0009c4b0 f8 ff ff 1a         bne    LAB_0009c498
```
**YOU MUST ALSO** modify the bytes on address 0x9c548 to be: `1c 4d 43 00`

If that's not what you want to do you may give me some more detail... in any case, if you want to load the contents of statusMqtt (0x434dc1) into r3 the instruction should be:
`ldr r3,[0x9c548]` along with modifying the bytes in 0x9c548 to be `1c 4d 43 00` (as I did with r0 above).

If you wanted to just set the variable from an outside script I assume that trick with echo+dd would work (as you previously confirmed):
`echo -en '\x01\x00\x00\x00' | dd of=/proc/$PID/mem bs=1 count=4 seek=$((0x434d1c)) `

**Comment by guino** [comment]
@jonesMeUp some other piece of information which can help.. apparently this can prevent the watchdog from resetting the device:
`(while true; do sleep 5; echo -n 1 > /dev/watchdog; done ) < /dev/null >& /dev/null &`
(basically sending `1` to /dev/watchdog every 5 seconds)

I kept my device running for a few minutes without a reboot and without anyka_ipc running that way -- however: the biggest problem is that anyka_ipc is what configures/connects the wifi on the device (so you can sync the time) so you'd have to start and configure the wifi by hand in order to get the time before running anyka_ipc. Still could be an option if all else fails.

**Comment by jonesMeUp** [comment]
@guino thx for your support!
You misunderstood me. I used your hint to set _statusMQTT_ in `hack.sh` as soon as ntpd has done its job.

What i want ito do:
### Step 1:
```
while (statusMQTT != 1) {
   sleep(1);
 }
```
Place this loop at _0x00065bc8_ in `void ht_log_init(void)` thats in `main()` **_NOT_** in `tuya_main()`
There is enough "empty" space for this loop and i want to test it for sideeffects.

### Step 2:
If this will work, i want to overwrite the loop in **tuya_main()** with a call to `ht_log_init(void)`
(The original caller function to ht_log_init(void) will be NOP out for the testing, later it can be "repaired")
At this position the wlan should be working and the time sync shouldn't be started.
This should work on both versions: your online and mine offline version.
```
while (statusMQTT != 1) {
   sleep(1);
 }
```

But i don't get the ldr to work. I now it holds a reference to an address, but i don't have a clue how
to calculate the offset. I asked the Duck, but got only topics where you must know already assembler to understand it.

**Comment by McPrapor** [comment]
Meanwhile update 2.10.36 is available again. At least two of my three devices silently updated to it, the third is still 2.10.28 though. Tomorrow will unpack the third one and check if it's available through the app.

**Comment by guino** [comment]
@jonesMeUp there's not a lot to it:

READ data from any address:
```
ldr r0,[0xPOINTER_ADDRESS]
ldr r3,[r0,#0x0]
```

WRITE data to any address:
```
ldr r0,[0xPOINTER_ADDRESS]
str r3,[r0,#0x0]
```

In both cases POINTER_ADDRESS will be an address containing the address to read/write, so if you wanted to read/write to 0x434d1c you would have to place the bytes `1c 4d 43 00` somewhere (near the code) and call that 'somewhere' POINTER_ADDRESS. You could even take out part of a string to write those bytes. In my example above https://github.com/guino/Merkury1080P/issues/42#issuecomment-1360832460 I used 0x9c548 because it was already storing some address used by code I overwrote. There's no 'offset calculation' required for this method, just requires the 8 bytes (for the 2 instructions) + 4 bytes to store the absolute address to read/write (but has to be near the code). 

So you'll need to copy all the bytes from the loop code to 0x065bc8 (assuming you have space) then you'll need some other 4 bytes (near that code) to put the address bytes `1c 4d 43 00`. It could look like this:

```
LOOP:   01 00 a0 e3     mov        r0,#0x1
        18 de fe eb     bl         <EXTERNAL>::sleep
        a0 30 9f e5     ldr        r3,[POINTER_ADDRESS]
        00 30 93 e5     ldr        r3,[r3,#0x0]
        01 00 53 e3     cmp        r3,#0x1
        f8 ff ff 1a     bne        LOOP
        ?? ?? ?? ea     b          EXIT_LOOP
POINTER_ADDRESS:
        1c 4d 43 00
EXIT_LOOP:
       <existing code>
```
(notice the 'missing' add instruction since I'm using absolute address defined in the POINTER_ADDRESS memory location)

You also have the option to use a few more instructions to build the address directly:
```
LOOP:   01 00 a0 e3     mov    r0,#0x1
        18 de fe eb     bl     <EXTERNAL>::sleep
        1c 30 a0 e3     mov    r3,#0x1c
        4d 3c 83 e3     orr    r3,r3,#0x4d00
        43 38 83 e3     orr    r3,r3,#0x430000
        00 30 13 e5     ldr    r3,[r3,#0x0]
        01 00 53 e3     cmp    r3,#0x1
        f9 ff ff 1a     bne    LOOP
```
Either one should work as long as you have enough space -- chances are whatever code you're overwriting/deleting (to put your code in place) already uses some address nearby that you can use for a pointer_address making it simpler to write without adding jumps or multiple instructions (seems like 0x65ca8 is the perfect POINTER_ADDRESS for you).

**Comment by guino** [comment]
> I thought 2.10.36 is available again, but looks like it's just the hack makes it believe it's running 2.10.36.

Yes, the hack uses a main application 2.10.36 so it reports that version to the server. I never did look into changing that since it works fine, but it could prevent us from knowing when a 2.10.36 is actually available for that device. Chances are it may come out as a different version than 2.10.36 anyway.

**Comment by jonesMeUp** [comment]
@guino i used your `build the address directly` version for my testings, worked well. I also added much `nop` instructions for you to give us more nice functions ;)
Still no success on localtime, but added `current lumi: x` to the event loop. I want to use the camera to switch on a light when its dark on movement event. Currently i use a _Sun_RiseSet_Timer_ calculation for this, but its not very accurate.
I hope to find more time for my testings, now that the holidays are nearly over.

**Comment by jonesMeUp** [comment]
`date` and `hwclock` now shows the correct date/time, so no more 1970 as root of the streaming folder.
But the video-stream still differs 1h. I will spend some more time on it, but if it tooks to much time - i will give up and live with it.

I will post my actual hack.sh and the patch file as soon as the 1h problem is fixed (or if i get lost in the fix).

**Comment by guino** [comment]
@jonesMeUp Nice to hear you've made progress. I recommend you try a normal boot (i.e. no SD card) to verify the video time is correct -- if it's off by 1h without the SD card then the issue isn't related to your changes (likely just some other setting).

**Comment by jonesMeUp** [comment]
@guino of course the problem is related to the offline patch. There are many functions that are only called when the device is online. 
To get the date fixed for the streaming folder i had to rewrite the `IPC_APP_Sync_Utc_Time()` funtion (which didn't make any sense in the offline version). 
The timestamp was coming from tuya server as a json string, now `dd` writes the timestamp into the new funtion after the time was synced.

I have already seen a possible location for the 1h problem:
```
[01-10 04:55:36-- TUYA Debug][gw_intf.c:4405] timezone:+01:00
```
The lines of the logfile are crying in my head, so i have to spend **some** time there.

**Comment by jonesMeUp** [comment]
@guino after searching for endless hours in the code, without any success, i think they are cheating.
Can you please check if `date` and `hwclock` differ, on your online cam, also to the vlc time display?

**Comment by guino** [comment]
@jonesMeUp sorry it's been a busy few days here.

This is my doorbell (2.9.6 fw):
```
~ # date
Tue Jan 17 14:24:31 CST 2023
~ # hwclock
-sh: hwclock: not found
~ # /mnt/mmc01/busybox hwclock
hwclock: can't open '/dev/misc/rtc': No such file or directory
~ # ls -la /etc/TZ
-rwxr-xr-x    1 wagner   root            12 Jan 17 02:07 /etc/TZ
~ # cat /etc/TZ
CST05:00:00
```

This is my indoor dumb camera (2.7.6 fw):
```
~ # date
Tue Jan 17 14:26:38 CST 2023
~ # hwclock
-sh: hwclock: not found
~ # /mnt/mmc01/busybox hwclock
hwclock: can't open '/dev/misc/rtc': No such file or directory
~ # ls -la /etc/TZ
-rwxr-xr-x    1 wagner   root            12 Jan 17 01:04 /etc/TZ
~ # cat /etc/TZ
CST05:00:00
```

This is my rotating camera (2.10.36):
```
~ # date
Tue Jan 17 09:29:56 GMT 2023
~ # hwclock
Tue Jan 17 09:29:57 2023  0.000000 seconds
~ # ls -la /etc/TZ 
lrwxrwxrwx    1 1000     1000             7 Aug 24  2021 /etc/TZ -> /tmp/TZ
~ # ls -la /tmp/TZ
-rw-rw-rw-    1 root     root            10 Jan 17 09:27 /tmp/TZ
~ # cat /tmp/TZ
GMT+05:00
```

My local time was 14:24-29 when testing the above -- please notice I am on Eastern US timezone (despite it showing CST), but all my devices show the correct time on the video.

**Comment by jonesMeUp** [comment]
@guino  so i searched many hours in my patches, but the `gmt/date/hwclock` "glitch" was in the original app already.
After you confirmed my suspicion, i updated my post to my actual version, showing the correct time only in OSD (like the original version). https://github.com/guino/Merkury1080P/issues/42#issuecomment-1353850449

**Comment by Pubagee** [comment]
Why are you doing this you pervert, some people use theese cameras for home security, i can tell you never grew up and never moved from your moms basement from your profile picture, you are like 40 years old you pervert, get a grip.

**Comment by usernamepaul** [comment]
Cyberduck
ftp
Server (192.168.xxx.xx)
port 21
user: root
pass:telnet

**Comment by der3b** [comment]
> 
> I will use the above information to leverage our options if it works on 2.10.36 and/or newer.

Hello Guino, works on fw 3.167.94.14. just got telnet access:
[root@anyka /usr/share]$ cat VERSION
3.167.94.14
[root@anyka /usr/share]$ cat BUNDLE
HT_IPC167KM_TUYA_AK3918EV330

**Comment by VirusBurst** [comment]
> I will use the above information to leverage our options if it works on 2.10.36 and/or newer.

Sadly now it doesn't work anymore :( Get this message in a loop
```
hwclock -w
echo GMT-00:00 > /tmp/TZ
Segmentation fault
*********************************
The MainApp was dead, restart it!
call: sh /usr/sbin/anyka_ipc.sh restart
restart ipc service......
killall: anyka_ipc: no process killed
stopping ipc service......
start ipc service......

```

BUNDLE: HT_IPC286KSD3M_TUYA_AK3918EV330
VERSION: 3.2863.105.28

But it still works to create the _ht_av_tuning.conf and shadow file to get via UART access to the system. 
Telnet doesn't work because

```
Failed to connect to non-global ctrl_ifname: wlan0  error: No such file or directory
wlan0     No such device
```

but I managed to copy mtdblock0 to 7 to SD card

[LSC_1296ptz.zip](https://github.com/user-attachments/files/21336261/LSC_1296ptz.zip)
